seq_logger.js

```diff
@@ -284,78 +275,47 @@ class SeqLogger {
 
         return new Promise((resolve, reject) => {
             const sendRequest = (batch, bytes) => {
+                const controller = new AbortController()
                 attempts++;
-                let req = this._httpModule.request({
-                    host: this._endpoint.hostname,
-                    port: this._endpoint.port,
-                    path: this._endpoint.path,
-                    protocol: this._endpoint.protocol,
-                    agent: this._httpAgent,
-                    headers: {
-                        "Content-Type": "application/json",
-                        "X-Seq-ApiKey": this._apiKey ? this._apiKey : null,
-                        "Content-Length": bytes,
-                    },
-                    method: "POST",
-                    timeout: this._requestTimeout
-                });
-
-                req.on("socket", (socket) => {
-                    if (socket.listeners("timeout").length == 0) {
-                        socket.on("timeout", () => {
-                            req.destroy();
-                            if (attempts > this._maxRetries) {
-                                return reject('HTTP log shipping failed, reached timeout (' + this._requestTimeout + ' ms)')
-                            } else {
-                                return setTimeout(() => sendRequest(batch, bytes), this._retryDelay);
-                            }
-                        });
-                    }
-                });
-
-                req.on('response', res => {
-                    var httpErr = null;
-                    if (res.statusCode !== 200 && res.statusCode !== 201) {
+                const timerId = setTimeout(() => {
+                  controller.abort()
+                  if (attempts > this._maxRetries) {
+                    reject('HTTP log shipping failed, reached timeout (' + this._requestTimeout + ' ms)')            
+                  } else {
+                    setTimeout(() => sendRequest(batch, bytes), this._retryDelay);
+                  }
+                }, this._requestTimeout)
+
+                fetch(this._endpoint, {
+                  keepalive: true,
+                  method: "POST",
+                  headers: {
+                    "Content-Type": "application/json",
+                    "X-Seq-ApiKey": this._apiKey ? this._apiKey : null,
```

Comment:
In my testing this is not working properly. `X-Seq-ApiKey: null` causes this error 'Payload from [...] specified invalid API key ·····'

Reply:
How did you test it?
Do you guys using your test-suits?